rest/server_context_test.go

```diff
@@ -163,3 +165,197 @@ func (client *MockClient) RespondToGET(url string, response *http.Response) {
+	return append(s[:index], s[index+1:]...)
+}
+
+func removeElementWithValue(s []string, e string) []string {
+	return removeElementWithIndex(s, indexOf(s, e))
+}
+
+func indexOf(elements []string, e string) int {
+	for k, v := range elements {
+		if e == v {
+			return k
+		}
+	}
+	return -1 // Not found.
+}
+
+func captureOutput(f func()) string {
+	var buf bytes.Buffer
+	log.SetOutput(&buf)
+	defer func() { log.SetOutput(os.Stderr) }()
+	f()
+	return buf.String()
+}
+
+func TestStartReplicators(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyReplicate)()
+	var replications []*ReplicationConfig
+
+	// Should be skipped; create_target option is not currently supported.
+	replications = append(replications, &ReplicationConfig{
+		Source:       "http://127.0.0.1:4985/imdb_us",
+		Target:       "http://127.0.0.1:4985/imdb_uk",
+		Async:        true,
+		CreateTarget: true})
+
+	// Should be skipped; doc_ids option is not currently supported
+	replications = append(replications, &ReplicationConfig{
+		Source: "http://127.0.0.1:4985/imdb_us",
+		Target: "http://127.0.0.1:4985/imdb_uk",
+		Async:  true,
+		DocIds: []string{"doc1", "doc2", "doc3", "doc4"}})
+
+	// Should be skipped; proxy option is not currently supported.
+	replications = append(replications, &ReplicationConfig{
+		Source: "http://127.0.0.1:4985/imdb_us",
+		Target: "http://127.0.0.1:4985/imdb_uk",
+		Async:  true,
+		Proxy:  "http://127.0.0.1:443/proxy"})
+
+	// Start a one-shot replication.
+	replications = append(replications, &ReplicationConfig{
+		Source:        "http://127.0.0.1:4985/imdb_us",
+		Target:        "http://127.0.0.1:4985/imdb_uk",
+		Async:         true,
+		ReplicationId: "58a632bb8d7e110445d3d65a98365d62"})
+
+	serverConfig := &ServerConfig{AdminInterface: &DefaultAdminInterface, Replications: replications}
+	serverContext := NewServerContext(serverConfig)
+	out := captureOutput(serverContext.startReplicators)
+
+	assert.Contains(t, "/_replicate create_target option is not currently supported", out)
+	assert.Contains(t, "/_replicate doc_ids option is not currently supported", out)
+	assert.Contains(t, "/_replicate proxy option is not currently supported", out)
+	assert.Contains(t, "ReplicationId:58a632bb8d7e110445d3d65a98365d62", out)
+}
+
+func TestPostStartup(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyReplicate)()
+	var replications []*ReplicationConfig
+
+	// Start a one-shot replication.
+	replications = append(replications, &ReplicationConfig{
+		Source:        "http://127.0.0.1:4985/imdb_us",
+		Target:        "http://127.0.0.1:4985/imdb_uk",
+		Async:         true,
+		ReplicationId: "58a632bb8d7e110445d3d65a98365d62"})
+
+	serverConfig := &ServerConfig{AdminInterface: &DefaultAdminInterface, Replications: replications}
+	serverContext := NewServerContext(serverConfig)
+	out := captureOutput(serverContext.startReplicators)
+	assert.Contains(t, "ReplicationId:58a632bb8d7e110445d3d65a98365d62", out)
```

Comment:
Again, capturing log output is probably not the right approach. It's something that should perhaps be used as an absolute last resort (or for testing logging specifically).

We can use the replicator endpoint to query the status of the given replication ID.

Reply:
This has been revised with `serverContext.replicator.ActiveTasks()`